net-misc/zerotier/zerotier-1.4.6.ebuild

```diff
@@ -0,0 +1,84 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit flag-o-matic llvm systemd toolchain-funcs
+
+HOMEPAGE="https://www.zerotier.com/"
+DESCRIPTION="A software-based managed Ethernet switch for planet Earth"
+SRC_URI="https://github.com/zerotier/ZeroTierOne/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="BSL-1.1"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~x86"
+IUSE="clang doc neon"
+
+S="${WORKDIR}/ZeroTierOne-${PV}"
+
+RDEPEND="
+	dev-libs/json-glib:=
+	net-libs/libnatpmp:=
+	net-libs/miniupnpc:=
+	clang? ( >=sys-devel/clang-6:= )"
+
+DEPEND="${RDEPEND}"
+
+PATCHES=( "${FILESDIR}/${P}-respect-ldflags.patch"
+	"${FILESDIR}/${P}-add-armv7a-support.patch"
+	"${FILESDIR}/${P}-fixup-neon-support.patch" )
+
+DOCS=( README.md AUTHORS.md )
+
+LLVM_MAX_SLOT=9
+
+llvm_check_deps() {
+	if use clang ; then
+		if ! has_version --host-root "sys-devel/clang:${LLVM_SLOT}" ; then
+			ewarn "sys-devel/clang:${LLVM_SLOT} is missing! Cannot use LLVM slot ${LLVM_SLOT} ..."
+			return 1
+		fi
+
+		if ! has_version --host-root "=sys-devel/lld-${LLVM_SLOT}*" ; then
+			ewarn "=sys-devel/lld-${LLVM_SLOT}* is missing! Cannot use LLVM slot ${LLVM_SLOT} ..."
+			return 1
+		fi
+
+		einfo "Will use LLVM slot ${LLVM_SLOT}!"
+	fi
+}
+
+src_compile() {
+	if use clang && ! tc-is-clang ; then
+		export CC=${CHOST}-clang
+		export CXX=${CHOST}-clang++
```

Comment:
Truthfully I'd just make a proper `pkg_setup` check with `|| die` guards instead of this kind of forcing. I can build this using clang and USE="-clang" with weird settings, right?

Doesn't this program have a configure step which checks your compiler?

Reply:
It doesn't have a configure phase per se; it is all handled via make.

Reply:
I think it's fine like this.